ports/mimxrt/led.c

```diff
@@ -0,0 +1,193 @@
+        IOMUXC_SetPinConfig(led_pin->muxRegister, 5U, 0, 0, led_pin->configRegister, 0x10B0U);
+        MICROPY_HW_LED_OFF(led_pin);
+    }
+}
+
+void led_state(machine_led_t led, int state) {
+    if (led < 1 || led > NUM_LEDS) {
+        return;
+    }
+
+    const pin_obj_t *led_pin = machine_led_obj[led - 1].led_pin;
+
+    if (state == 0) {
+        // turn LED off
+        MICROPY_HW_LED_OFF(led_pin);
+    } else {
+        // turn LED on
+        MICROPY_HW_LED_ON(led_pin);
+    }
+}
+
+void led_toggle(machine_led_t led) {
+    if (led < 1 || led > NUM_LEDS) {
+        return;
+    }
+
+    const pin_obj_t *led_pin = machine_led_obj[led - 1].led_pin;
+    mp_hal_pin_toggle(led_pin);
+}
+
+void led_debug(int value, int delay) {
+    for (mp_int_t i = 0; i < NUM_LEDS; i++) {
+        led_state(i + 1, (value & (1 << i)));
+    }
+    mp_hal_delay_ms(delay);
+}
+
+
+/******************************************************************************/
+/* MicroPython bindings                                                       */
+
+STATIC void led_obj_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    (void)kind;
+    machine_led_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_printf(print, "LED(%u)", self->led_id);
+}
+
+/// \classmethod \constructor(id)
+/// Create an LED object associated with the given LED:
+///
+///   - `id` is the LED number, 1.
+STATIC mp_obj_t led_obj_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    mp_arg_check_num(n_args, n_kw, 1, 1, false);
+
+    // Extract arguments
+    mp_int_t led_id = mp_obj_get_int(args[0]);
+
+    // Check led id is in range
+    if (!(1 <= led_id && led_id <= NUM_LEDS)) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError, "LED(%d) doesn't exist", led_id));
+    }
+
+    // Return reference to static object
+    return MP_OBJ_FROM_PTR(&machine_led_obj[led_id - 1]);
+}
+
+/// \method on()
+/// Turn the LED on.
+STATIC mp_obj_t led_obj_on(mp_obj_t self_in) {
+    machine_led_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    MICROPY_HW_LED_ON(self->led_pin);
+    return mp_const_none;
+}
+
+/// \method off()
+/// Turn the LED off.
+STATIC mp_obj_t led_obj_off(mp_obj_t self_in) {
+    machine_led_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    MICROPY_HW_LED_OFF(self->led_pin);
+    return mp_const_none;
+}
+
+/// \method toggle()
+/// Toggle the LED between on and off.
+STATIC mp_obj_t led_obj_toggle(mp_obj_t self_in) {
+    machine_led_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_hal_pin_toggle(self->led_pin);
+    return mp_const_none;
+}
+
+// TODO: Implement intensity method
+/// \method intensity([value])
+/// Get or set the LED intensity.  Intensity ranges between 0 (off) and 255 (full on).
+/// If no argument is given, return the LED intensity.
+/// If an argument is given, set the LED intensity and return `None`.
+mp_obj_t led_obj_intensity(size_t n_args, const mp_obj_t *args) {
+    return mp_const_none;
+}
```

Comment:
I'd say just leave this unimplemented function out for now.  That way the code raises an exception when trying to access it, rather than failing silently.